pkg/cmd/services_test.go

```diff
@@ -442,11 +455,128 @@ func TestServicesCmd_CreateServiceInstanceCmd(t *testing.T) {
 			if tc.ValidateErr != nil {
 				tc.ValidateErr(t, err)
 			}
+		})
+	}
+}
 
+func TestServicesCmd_ListServiceInstanceCmd(t *testing.T) {
+	cases := []struct {
+		Name             string
+		SvcCatalogClient func() versioned.Interface
+		K8Client         func() kubernetes.Interface
+		ExpectError      bool
+		ValidateErr      func(t *testing.T, err error)
+		ValidateOut      func(t *testing.T, output []byte)
+		Args             []string
+		Flags            []string
+	}{
+		{
+			Name: "test list service instances as expected",
+			SvcCatalogClient: func() versioned.Interface {
+				fake := &scFake.Clientset{}
+				fake.AddReactor("list", "serviceinstances", func(action ktesting.Action) (handled bool, ret runtime.Object, err error) {
+					return true, &v1beta1.ServiceInstanceList{
+						Items: []v1beta1.ServiceInstance{
+							{
+								ObjectMeta: metav1.ObjectMeta{
+									GenerateName: "keycloak",
+									Name:         "keycloak",
+									Labels: map[string]string{
+										"serviceName": "keycloak",
+									},
+								},
+							},
+						},
+					}, nil
+				})
+				return fake
+			},
+			K8Client: func() kubernetes.Interface {
+				return &kFake.Clientset{}
+			},
+			ValidateOut: func(t *testing.T, data []byte) {
+				var list = &v1beta1.ServiceInstanceList{}
+
+				if err := json.Unmarshal(data, list); err != nil {
+					t.Fatal("failed to unmarshal data", err)
+				}
+				if nil == list {
```

Comment:
I don't think list would ever be nil here as it has been allocated on line 498 so can probably remove this

Reply:
Good point, thanks